src/rpc/mining.cpp

```diff
@@ -544,6 +544,7 @@ UniValue getblocktemplate(const JSONRPCRequest& request)
         pindexPrev = pindexPrevNew;
     }
     CBlock* pblock = &pblocktemplate->block; // pointer for convenience
+    assert(pblock);
```

Comment:
It's impossible for pblock to be NULL here. The dereference is one line above. And pindexPrev initialises to NULL, so the above initialisation will indeed always occur.